src/modules/sway/hide.cpp

```diff
@@ -0,0 +1,52 @@
+#include "modules/sway/hide.hpp"
+#include <spdlog/spdlog.h>
+#include "client.hpp"
+#include "wlr-layer-shell-unstable-v1-client-protocol.h"
+
+namespace waybar::modules::sway {
+
+Hide::Hide(const std::string& id, const Bar& bar, const Json::Value& config)
+    : ALabel(config, "hide", id, "{}", 0, true), bar_(bar), windowId_(-1) {
+  ipc_.subscribe(R"(["bar_state_update","barconfig_update"])");
+  ipc_.signal_event.connect(sigc::mem_fun(*this, &Hide::onEvent));
+  // Do not reserve space for the bar anymore
+  bar.removeExclusiveZone();
+  // Launch worker
+  worker();
+}
+
+void Hide::onEvent(const struct Ipc::ipc_response& res) {
+  auto payload = parser_.parse(res.payload);
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (payload.isMember("mode")) {
+    // barconfig_update: get mode
+    auto mode = payload["mode"].asString();
+    if (mode == "hide") {
+      // Hide the bars when configuring the "hide" bar
+      for (auto& bar : waybar::Client::inst()->bars) {
+        bar->setVisible(false);
+      }
+    }
```

Comment:
With swaybar, users can easily toggle whether their bar is in `hide` mode or `dock` (normal) mode.
```
bindsym $binding bar mode toggle
```
This alternates between `dock` and `hide` mode. When in `hide` mode, we should:

- Set the layer to `bottom`
- Remove the exclusive zone

When in `dock` mode, we should:

- Set the layer to the user-configured layer
- Re-enable the exclusive zone

Reply:
I merged your fork but replaced the functions in favor of @alebastr suggestion.